### DISCUSSION TITLE
Is there any retry backoff implementation?

### DISCUSSION THREAD

**Discussion by james-ryans** [linkerd/linkerd2#11194] [General]
Is there any retry backoff implementation?
If yes, what is the default backoff and how to configure it? Is there any exponential backoff with jitter implementation?
If no, so Linkerd will retry exactly after the request has failed? Why is there no such feature? Can I have the reasoning for not implement this?

**Comment by wmorgan** [comment]
Linkerd features retry budgets which have better worst-case behavior than backoff. Please see https://linkerd.io/2.13/tasks/configuring-retries/

**Comment by james-ryans** [comment]
Yes, I've read the retry configurations and did some testing. May I have the reasoning for not implementing backoff even though there's a retry budget feature? In my experiment, I did a load test to emojivoto vote doughnut which is a 100% failure rate, and with 10s ttl, the Linkerd bombardier the app with retries at 0th second, 10th second, 20th second, and so on.
